pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -160,6 +161,8 @@
   private final ThreadSafeMutableRoaringBitmap _validDocIds;
   private final ValidDocIndexReader _validDocIndex;
 
+  private final PinotSegmentRecordReader _pinotSegmentRecordReader;
```

Comment:
I don't think we need to initialize a record reader within the mutable segment. How are you planning to use it?

Reply:
@Jackie-Jiang Got it. I think i am going to use this API in another PR. When lookup the previous record for merge, 
do you think it's suitable to make recordReader an instance variable of mutableSegment? or should I init recordReader a local variable of PartialUpsert Handler?